daemon/cluster/listen_addr.go

```diff
@@ -99,7 +99,7 @@ func validateDefaultAddrPool(defaultAddrPool []string, size uint32) error {
 	if size == 0 {
 		size = 24
 	}
-	if size > 32 {
+	if size > 29 {
```

Comment:
Could you add a comment, explaining why 29 is the maximum? (Is this to reserve space for the gateway address?)

Reply:
No. Its for how many IP addresses we will acquire for the newly created overlay network.   subnet mask length , we can give max 32.  But if we give 29 is the max value, we will be having at least 8 addresses.  if we give 30, we will be having 4 addresses.  So 29 seems reasonable value where we will be getting at least 8 addresses.

Reply:
Not sure I understand; if a user would _want_ a smaller size, why do we want to prevent them doing so?

Reply:
There was a discussion in the issue thread (https://github.com/docker/docker-core-backlog/issues/433) where it was discussed "smallest usable docker overlay network is 8 addresses.".  I was going through the same discussion with team and we decided to keep 8 as max number. But if you think any user will 4 addresses overlay network, then we can change it to 30.  But I feel thats most unlikely. WDYT?

Reply:
That internal issue mentions;

> Using 30 here produces a pool of network with four addresses in each network. Since all four addresses are allocated before the container attaches, this is also results in an unusable pool.

What allocates those four addresses? Is that only in a Docker Enterprise / UCP situation?

I understand that having less than 8 (or 4 even) addresses per pool is an _unlikely_ scenario, but there may be use cases we're excluding here.

Reply:
I don't think its only EE/UCP scenario. We use min 4 IPs for our internal use if I am not worng.

Reply:
In older releases when I tried we used 3 IPs and I can attach one container. Then IP addresses exhaust if you try to attach another container.
 In new release due to LB scale feature , we use 4 ipaddresses and hence it doesnt allow to attach any container due to IP address exhaustion.

Reply:
Ok, so that makes sense; in that case the _technical_ limit is that we need to have 4 (reserved by default for LB) + 1 (for at least one container) in order to get a functional network.

That's important information to add to that code as a comment, otherwise `29` is just some magic number, without an explanation why we picked that number.

_Ideally_ we'd list (in that comment) what is taking those 4 IP-addresses, so that someone getting to that code in future understands how we got to the 4 reserved IP-addresses (without having to Git blame the line, or dig through Git history)

Reply:
added max information in the comment explaining why we choose 29 as max number.

Reply:
@thaJeztah @tiborvass @crosbymichael PTAL